### PR TITLE
feat: Redesign UI

### PR DESCRIPTION
**Pull request by StephixOne**

### CODE REVIEW

File: src/renderer/PovSelection.tsx [hunk 1]
```diff
@@ -125,26 +104,19 @@ export default function PovSelection(props: IProps) {
...
-      let onClick = (
-        event: React.MouseEvent<HTMLButtonElement, MouseEvent>
-      ) => {
-        handleChangePov(event, cloudIndex);
-      };
+      let title = 'Use cloud version';
 
       if (!haveCloudVideo) {
         opacity = 0.2;
         title = 'No cloud recording is saved';
```
Comment (aza547) [inline src/renderer/PovSelection.tsx:111]: Mousing over a disabled icon doesn't show this tooltip anymore.

File: src/renderer/RecorderPreview.tsx [hunk 2]
```diff
@@ -52,6 +52,7 @@ const RecorderPreview: React.FC = () => {
         border: '2px solid black',
         boxSizing: 'border-box',
       }}
+      className="z-0"
```
Comment (aza547) [inline src/renderer/RecorderPreview.tsx:55]: Does this make any difference?

File: src/renderer/VideoButton.tsx [hunk 3]
```diff
@@ -393,160 +283,119 @@ export default function VideoButton(props: IProps) {
...
-
   const getShareableLink = async (event: React.MouseEvent<HTMLElement>) => {
     event.stopPropagation();
     event.preventDefault();
 
     try {
       await ipc.invoke('getShareableLink', [videoName]);
-      setLinkSnackBarSuccessOpen(true);
+      toast({
+        title: 'Share link generated',
```
Comment (aza547) [inline src/renderer/VideoButton.tsx:306]: Wording not quite right here: "Link generated" / "Shareable link generated" would be fine. Although it's not totally obvious it's been copied to the clipboard so I think we want to get that information in here too.

**Review by aza547 — CHANGES_REQUESTED** [review]
I think the "can't change while recording" bit has been missed in the styling:
image
image
image
image

Resolution selection dropdown menu goes the wrong way:
image

A few other minor things throughout the diff and a few questions to clarify my understanding, I had a skim over the components in a bit less detail than the rest as there is an awful lot of css classes in there that are pretty hard to pick apart, but it looks great so I'm happy with that :)

File: src/renderer/App.tsx [hunk 4]
```diff
@@ -46,6 +52,27 @@ const WarcraftRecorder = () => {
...
+
+    if (upgradeStatus.available) {
+      toast({
+        title: 'Update available!',
+        description:
+          'There is an update available for Warcraft Recorder. Please click the button below to download it.',
+        action: (
+          <a href={upgradeStatus.link} className="w-full">
+            <ToastAction altText="Download">
+              <ArrowBigDownDash /> Download
```
[the remark this answers is not in the document]
Comment (StephixOne) [reply src/renderer/App.tsx:66]: Good shout, I didn't intentionally change this, think it was something I hadn't thought through when prototyping the new upgrade button/notification. I do think it's better for it to go via browser, I'll change

[the remark this answers is not in the document]
Comment (StephixOne) [reply src/renderer/App.tsx:66]: Updated both instances 👌

File: src/renderer/ChatOverlayControls.tsx [hunk 5]
```diff
@@ -92,279 +47,212 @@ const ChatOverlayControls: React.FC = () => {
...
+          Scale
+        </Label>
+        <div className="flex w-48 h-[20px] items-center">
+          <Slider
+            defaultValue={[config.chatOverlayScale]}
+            value={[config.chatOverlayScale]}
+            max={5}
+            step={0.05}
+            onValueChange={setScale}
+            disabled={!config.chatOverlayEnabled}
```
[the remark this answers is not in the document]
Comment (StephixOne) [reply src/renderer/ChatOverlayControls.tsx:252]: Sure, all sliders gone when chat overlay disabled.

File: src/renderer/AudioSourceControls.tsx [hunk 6]
```diff
@@ -195,255 +121,199 @@ const AudioSourceControls: React.FC = () => {
...
-            sx={sliderSx}
-            value={config.speakerVolume * 100}
-            onChange={setSpeakerVolume}
-          />
-          <VolumeUp sx={{ color: 'white' }} />
-        </Stack>
-      </Box>
+      <div className="w-full flex gap-x-2 items-center">
+        <Volume1 />
+        <Slider
```
[the remark this answers is not in the document]
Comment (StephixOne) [reply src/renderer/AudioSourceControls.tsx:189]: Not easily doable without making the slider thumb bigger. Figure this isn't a frequent enough interaction for this to cause too much friction, but could explore a fix in a future version.

File: src/renderer/Layout.tsx [hunk 7]
```diff
@@ -49,13 +71,24 @@ const Layout = (props: IProps) => {
...
   } = props;
   const { page, category } = appState;
+  const [appVersion, setAppVersion] = React.useState<string>();
 
-  const handleChangeCategory = (
-    _: React.SyntheticEvent,
-    newCategory: VideoCategory
-  ) => {
+  React.useEffect(() => {
+    window.electron.ipcRenderer.on('updateTitleBar', (t: unknown) => {
```
[the remark this answers is not in the document]
Comment (StephixOne) [reply src/renderer/Layout.tsx:84]: Renamed

File: src/renderer/PovSelection.tsx (hunk 1 above)
Comment (StephixOne) [reply src/renderer/PovSelection.tsx:111]: Made them show with some tailwind magic.

File: src/renderer/RecorderPreview.tsx (hunk 2 above)
Comment (StephixOne) [reply src/renderer/RecorderPreview.tsx:55]: Oh, no, left over from when I was trying to get this preview to be less problematic 😀

File: src/renderer/VideoButton.tsx [hunk 8]
```diff
@@ -393,160 +283,119 @@ export default function VideoButton(props: IProps) {
...
-            width: '35%',
-            display: 'flex',
-            alignItems: 'center',
-            justifyContent: 'center',
-          }}
-        >
+        <div className="h-full w-1/3 flex items-center content-center">
           <PovSelection
-            povs={povs}
+            povs={[...povs, ...povs, ...povs, ...povs]}
```
[the remark this answers is not in the document]
Comment (StephixOne) [reply src/renderer/VideoButton.tsx:391]: Yes, my bad, didn't catch it when pushing. It didn't have the effect I wanted to I didn't notice it still in 😀

File: src/renderer/VideoButton.tsx (hunk 3 above)
Comment (StephixOne) [reply src/renderer/VideoButton.tsx:306]: Changed toast title to `Shareable link generated and placed in clipboard`

File: src/renderer/components/Dialog/Dialog.tsx [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+
+const DialogClose = DialogPrimitive.Close;
+
+const DialogOverlay = React.forwardRef<
+  React.ElementRef<typeof DialogPrimitive.Overlay>,
+  React.ComponentPropsWithoutRef<typeof DialogPrimitive.Overlay>
+>(({ className, ...props }, ref) => (
+  <DialogPrimitive.Overlay
+    ref={ref}
+    className={cn(
```
[the remark this answers is not in the document]
Comment (StephixOne) [reply src/renderer/components/Dialog/Dialog.tsx:22]: First it runs `clsx` on its input, which lets me write conditional class names in a neat fashion, you'll see some examples in the code, but basically you can do something like `className={cn('static classes', { 'conditional classes': condition })}`, then runs `tailwind-merge` on the result of this, which is just a nice utility to ensure we get a string of tailwind classes that "makes sense" and avoids style conflicts.

File: src/renderer/App.css [hunk 10]
```diff
@@ -2,13 +2,90 @@
  * @NOTE: Prepend a `~` to css file paths that are in your node_modules
  *        See https://github.com/webpack-contrib/sass-loader#imports
  */
+@import url('https://fonts.googleapis.com/css2?family=Inter:opsz,wght@14..32,100..900&display=swap');
```
[the remark this answers is not in the document]
Comment (aza547) [reply src/renderer/App.css:5]: We chatted in discord about this and agreed it's fine - this API is well used and reliable.

File: src/renderer/AudioSourceControls.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (aza547) [reply src/renderer/AudioSourceControls.tsx:189]: OK :)

**Review by aza547 — APPROVED** [review]
Thanks again for all the hard work here - this looks really great.

**Pull request merged**